### PR TITLE
Refactor button reservable from another library

### PR DESCRIPTION
**Pull request by kasperbirch1**
#### Link to issue
https://reload.atlassian.net/browse/DDFSOEG-528

#### Description

This pull request refactors the order buttons by moving the logic for displaying the `MaterialButtonReservableFromAnotherLibrary` component and refactoring the `getReservablePidsFromAnotherLibrary` function into a `useReservableFromAnotherLibrary` hook. That also checks if the material is not reservable and should be ordered via openOrder.

#### Screenshot of the result

### CODE REVIEW

File: src/core/utils/useReservableFromAnotherLibrary.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { useConfig } from "./config";
+import { getAllFaustIds } from "./helpers/general";
+import { useGetHoldings } from "../../apps/material/helper";
+import { Manifestation } from "./types/entities";
+
+const useReservableFromAnotherLibrary = (manifestations: Manifestation[]) => {
```
Comment (spaceo) [inline src/core/utils/useReservableFromAnotherLibrary.tsx:6]: Great with the hook.
I think tou should add some tests testing the hook now that you have deleted the old tests.

**Review by spaceo — CHANGES_REQUESTED** [review]
Looks good. But I think it is a bad idea to delete tests without adding new

**Review by spaceo — APPROVED** [review]
LGTM 👍

**Comment by kasperg** [comment]
@spaceo already gave this a :+1: :smile:

**Pull request merged**